### ISSUE TITLE
Vladmandic SDNext Issue RuntimeError: Cannot use secondary sampler PLMS with Dynamic Thresholding

### ISSUE THREAD

**Issue by Lucius338** [mcmonkeyprojects/sd-dynamic-thresholding#63]
Clean install of the Vladmandic fork installed today, but for some reason I cannot get dynamic thresholding to work for the life of me. Tried disabling all extensions, including Controlnet, no dice. I think it has to do with PLMS being set as the default sampler in this fork, and I can't find a way to change it in the config file. At first, I received the same error but it said "Cannot use secondary sampler DDIM with Dynamic Thresholding." Is one of the optimizations responsible for assigning a second sampler? 

16:44:43-323693 ERROR    Running script process batch:
                         C:\sdnext082023\extensions-builtin\sd-dynamic-thresholding\scripts\dynamic_thresholding.py:
                         RuntimeError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ C:\sdnext082023\modules\scripts.py:451 in process_batch                                                             │
│                                                                                                                     │
│   450 │   │   │   │   args = p.per_script_args.get(script.title(), p.script_args[script.args_f                      │
│ > 451 │   │   │   │   script.process_batch(p, *args, **kwargs)                                                      │
│   452 │   │   │   │   s.append(f'{script.title()}:{round(time.time()-t0, 2)}s')                                     │
│                                                                                                                     │
│ C:\sdnext082023\extensions-builtin\sd-dynamic-thresholding\scripts\dynamic_thresholding.py:99 in process_batch      │
│                                                                                                                     │
│    98 │   │   if orig_latent_sampler_name in ["DDIM", "PLMS"]:                                                      │
│ >  99 │   │   │   raise RuntimeError(f"Cannot use secondary sampler {orig_latent_sampler_name}                      │
│   100 │   │   if 'UniPC' in (orig_sampler_name, orig_latent_sampler_name) and p.enable_hr:                          │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
RuntimeError: Cannot use secondary sampler PLMS with Dynamic Thresholding

**Comment by mcmonkey4eva** [comment]
... Why the heck is PLMS default? 0.o that sampler is really weird, and has a lot of compatibility issues as you see here.

I'd recommend DPM++ 2M Karras as a go-to default for most usages (or Euler if you want the slower basic one that's highly replicable)

**Comment by Lucius338** [comment]
Yeah, idk why, I swear there was a way to change the default sampler in the config before but I can't find it anymore.

Anyways, I found the issue. I didn't even think about second pass because I hadn't been using it, but even with Second Pass disabled, if its sampler is set to default, it's enough to trip up Dynamic Thresholding. The workaround for now is manually changing the second pass sampler even if you're not using it. Should I close this topic or should there be a solution implemented for this?

**Comment by vladmandic** [comment]
> ... Why the heck is PLMS default? 0.o that sampler is really weird, and has a lot of compatibility issues as you see here.
> 
> ... Why the heck is PLMS default? 0.o that sampler is really weird, and has a lot of compatibility issues as you see here.

default in sdnext is whatever is set as default in a loaded model. sdnext doesn't choose plms itself.
regardless of that, i suggest to change logic to fallback to primary sampler is selected latent sampler is not set to a compatible one - raising `RuntimeError` is not a user-friendly behavior.

slightly off-topic, unipc is also blacklisted by extension because it doesn't support img2img - that was the case long time ago, sdnext supports unipc fully for txt2img and img2img.
